pkg/crc/machine/machine.go

```diff
@@ -218,6 +219,25 @@ func Start(startConfig StartConfig) (StartResult, error) {
 		return *result, err
 	}
 
+	// kubeconfig file is also needed inside the VM to perform pull secret replace operation.
+	kubeConfigFile, err := ioutil.ReadFile(filepath.Join(constants.MachineInstanceDir, machineConfig.Name, "kubeconfig"))
+	if err != nil {
+		logging.ErrorF("Failed to read %s: %v", filepath.Join(constants.MachineInstanceDir, machineConfig.Name, "kubeconfig"), err)
+		result.Error = err.Error()
+		return *result, err
+	}
+
+	// If it is restart then don't again add user pull secret.
+	if !exists {
+		// Update the user pull secret before kubelet start.
+		logging.Info("Adding user's pull secret and cluster ID ...")
+		if err := pullsecret.AddPullSecretAndClusterID(host.Driver, startConfig.PullSecret, string(kubeConfigFile)); err != nil {
+			logging.ErrorF("Failed to update user pull secret: %v", err)
```

Comment:
@cfergeau it is the `err` variable which will give idea about what actually which command failed.

Reply:
Sorry, was a bit cryptic, but the error message is `Failed to update user pull secret` which is not correct as it could be the cluster id which we could not update.

Reply:
@cfergeau what do you suggest should be something like `Failed to update user pull secret or Cluster id` as message ?

Reply:
yes something like this :)

Reply:
This does not seem updated in the latest iteration.